elements/lisk-tree/src/merkle_tree/merkle_tree.ts

```diff
@@ -267,13 +276,100 @@ export class MerkleTree {
 		return rightWitness;
 	}
 
+	public async update(idxs: number[], updateData: ReadonlyArray<Buffer>): Promise<Buffer> {
```

Comment:
follow up on the previous comment: this is also an array of encoded indexes, maybe we could use the same variable name here (also I would at least use `indexes` here or `idxs` in `calculateRootFromUpdateData`)

Reply:
Furthermore, these idxs can only correspond to leaf nodes. If they are encoded, we can actually still check that they have the correct length. Thinking about it, I think this is important, because the real update is given by the `idxs` array, if they can correspond to internal nodes, we allow for updates of internal nodes.

Reply:
Added layer check